torch/distributed/fsdp/_init_utils.py

```diff
@@ -838,6 +839,14 @@ def _need_to_materialize_module(
     """
     managed_params = list(_get_orig_params(module, ignored_params))
     is_meta_module = any(param.is_meta for param in managed_params)
+    # TODO: We need to establish a contract for FSDP and buffers. For now, we
+    # skip checking for meta buffers from ignored modules. We should consider
```

Comment:
what about meta parameters from ignored modules?

Reply:
Yes, we need to discuss that too in terms of FSDP's contract. For now, meta parameters from ignored modules are also not considered (i.e. they will not make the module considered to need materialization).